eds/builders/report_builder.py

```diff
@@ -0,0 +1,54 @@
+""" 
+report_generator
+Created:  08/03/2019
+Author: Eugeniu Costetchi
+Email: costezki.eugen@gmail.com
+
+this module implements the report generation functionality.
+"""
+import jinja2
+
+from eds.builders.jinja_builder import build_eds_environment, inject_environment_globals
+
+
+class ReportBuilder:
+    """
+        generic report builder that takes templates and configuration as input and produces an output report
+    """
+
+    def __init__(self, template_folder, main_template_name, configuration_context=None):
+        """
+            Instantiates builders form a template providing an optional configuration context.
+
+        :param template_folder: the folder where JINJA2 templates are found
+        :param main_template_name: the main builders template that will be used to generate the builders
+        :param configuration_context: the configuration context available in the templates under the key "configuration"
+        """
+        self.main_template_name = main_template_name
+        self.configuration_context = configuration_context
+
+        template_loader = jinja2.FileSystemLoader(searchpath=str(template_folder))
+        self.template_env = build_eds_environment(loader=template_loader)
+        inject_environment_globals(self.template_env, {'configuration', self.configuration_context})
```

Comment:
what happens if `configuration_context` is `None`?

Reply:
it stays empty

Reply:
you're sending a `set` as the second parameter of `inject_environment_globals`. This method has no specification on the type of parameter, but the method seems to expect a `dict` (the `context.items()` call, `set` doesn't have a `.items()` method), so it will most likely throw an exception.
```python
def inject_environment_globals(jinja_environment, context, update_existent=True):
    """
        Inject the context into JINJA2 environment making it globally available from any template.
        Update in place a dictionary by adding non existent keys from another dictionary.
        If the dictionary keys exist then they are replaced depending whether the update_existent flag is set.
    :param update_existent:
    :param context:
    :param jinja_environment:
    :return:
    """
    if update_existent:
        jinja_environment.globals.update(context)
    else:
        jinja_environment.globals.update(
            {k: v for k, v in context.items() if k not in jinja_environment.globals.keys()})
```